lib/cloudant.js

```diff
@@ -49,11 +51,17 @@ function Cloudant(settings, ds) {
     throw new Error('Invalid settings: "url" OR "username"' +
     ' AND "password" required');
   }
-  this.creds = settings.url || {
-    account: settings.username,
-    password: settings.password,
-  };
-  this.cloudant = Driver(this.creds);
+
+  if (settings.url) {
+    this.options = settings.url;
```

Comment:
@kjdelisle From cloudant document I think they still expect arbitrary properties like `plugin` to be passed in separately, not inside `url`:
e.g: the example in https://github.com/cloudant/nodejs-cloudant#the-promises-plugins
`var cloudant = Cloudant({url: myurl, plugin:'promises'});`

so with `url` we still need `_.merge(this.options, settings);`.